app/models/penalty_decay_period_submission_rule.rb

```diff
@@ -1,13 +1,24 @@
 class PenaltyDecayPeriodSubmissionRule < SubmissionRule
   # This message will be dislayed to Students on viewing their file manager
   # after the due date has passed, but before the calculated collection date.
+  validates :penalty_type,
+            inclusion: { in: [ExtraMark::PERCENTAGE, ExtraMark::POINTS, ExtraMark::PERCENTAGE_OF_MARK] }
+
   def overtime_message(grouping)
     # How far are we into overtime?
     overtime_hours = calculate_overtime_hours_from(Time.current, grouping)
     # Calculate the penalty that the grouping will suffer
     potential_penalty = calculate_penalty(overtime_hours)
+    type = case penalty_type
```

Comment:
This is a bit more complicated than it needs to be. Write the `_` right after `overtime_message` below, so that you can use `#{penalty_type}` directly in the interpolation string.

This comment applies to the other submission rule below.